### DISCUSSION TITLE
Can't access public assets when deployed

### DISCUSSION THREAD

**Discussion by valerebron** [nuxt/nuxt#22761] [Questions]
Hi !
I deploy a Nuxt 3 app and I need to serve static files (external from the git repo) by public folder.

To do that I bind a docker volume like this :
``` yml
    volumes:
      - "${MEDIA_DIR}/music:/app/.output/public/music"  # for front access
      - "${MEDIA_DIR}/music:/app/public/music"          # for back access
```

Music/ folders & files permissions are 755, www-data:www-data on the host & root:root in the docker container.

The problem is, I can't access files from url domain.com/files.mp3 (404)
It works when I add files.mp3 directly in the public folder in the repo so there's no network miss-configuration.

Also when I dig inside the container app, I can list files.mp3 in the public folders.

So why nuxt cant serve these static files ?
Is there a better way to use external static files as assets in a nuxt 3 deployed app ?

**Comment by xjccc** [comment]
![image](https://github.com/nuxt/nuxt/assets/16478776/e548191a-2583-4f4c-9458-ba1e0e0d519f)

[docs](https://nuxt.com/docs/getting-started/assets#example-1)

is this?

**Reply by valerebron** [reply]
Hi Xjccc and thanks for your response !

I'm talking about public/ folder not assets/ folder.
I update my question to be more precise.

**Reply by xjccc** [reply]
oh, sorry, i dont know this.

i think add a cdnUrl  if its a way to have a try ?

**Comment by logotip4ik** [comment]
Nitro won't serve files in public folder, that were added after build step (https://github.com/nuxt/nuxt/issues/15779#issuecomment-1397418353). You could use something like `nitro:build:public-assets` (https://nuxt.com/docs/api/advanced/hooks) hook to add more assets. Another option would be custom server handler that serves files directly from file system (https://github.com/nuxt/nuxt/issues/15779#issuecomment-1397418381), but i would argue it's security

![joining paths](https://github.com/nuxt/nuxt/assets/56201308/5a546ed5-8abf-4d71-a705-31abaa726c25)

**Reply by valerebron** [reply]
Thanks Logotip4ik !
Finally, I build a custom server running on another domain.
I prefer to keep all app services in the same domain & container (for security & flexibility) but I got to heavy files to be copied in .output folder and for now I think I'm constrained to copied files in .output folder.
